### PR TITLE
Showcase functional updates

### PR DESCRIPTION
**Pull request by sertmer**
This PR closes issue #633 

### Changes Requested in Issue:

- [x] The rubric needs to be updated to expect the use of functional components, and the use of the useState and useEffect hooks to manage state and parse updates from the API.
- [x] Make it clear that a failure to participate can result in an individual failing the project, even if the project is otherwise passing
- [x] Update rubrics to use the Yes, Not Yet, Wow scale.

### Changes Made:

- [x] Rubric did not mention class based components previously, so no changes necessary there
- [x] Made clear that failure to participate can lead to a failing project
- [x] Rubric updated to wow, yes, and not yet

### CODE REVIEW

File: projects/module-3/showcase.md [hunk 1]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
+* A clear understanding of stateful components vs stateless components is demonstrated
+* Only the data that a child component _needs_ is passed down as props
+* Logic is kept out of return statements; return statements are as readable as possible, only communicating what will be displayed
+* The data displayed on the frontend is consistent with the data stored on the backend
+* Functions follow the single responsibility principle and are written to be reusable when appropriate
+* Data fetched from API is cleaned before being set to state
+* All components which receive props implement prop typechecking (proptypes or otherwise)  
+
+✨WOW✨ can look like:
+* Refactoring your stateful components into class-based components. Be sure to do this on a separate branch and submit the link to that PR to your Project Manager. Please do not merge it in. Highlight your ability to work with both styles by describing the refactor and linking to the branch in your README.
```
Comment (CassandraGoose) [inline projects/module-3/showcase.md]: Love the idea to have students learn this but not merge it in to their code, so they understand that it's not necessarily _better_ it's just another way of doing things. 🎉

File: projects/module-3/showcase.md [hunk 2]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
+* Commits are atomic and frequent, effectively documenting the evolution/progression of the application. There is no more than a 10% disparity in project contributions between teammates.
+* A project board is utilized (and updated throughout the project) with Github issues and labels.
+* Team uses branches, PRs and thorough code reviews to add new code to the main branch.
+* The README is formatted well and at a minimum contains:
+  * Overview of project and goals
+  * Overview of technologies used, challenges, wins, and any other reflections
+  * Screenshots/gifs of your app
+  * List of contributors
+* **Team collaborates effectively to accomplish the shared goal.  Team productively and professionally works through challenges and conflicts to ensure all team members are able to be heard and contribute throughout the project.**  
+  * Instructors are available to offer support and guidance but conversations around what *is* and what *is not* working are expected to be led by the team members themselves.
```
Comment (hfaerber) [inline projects/module-3/showcase.md]: Since this is a solo project we should be able to remove this section

**Review by CassandraGoose — APPROVED** [review]
Seems solid to me. 🙌 I would feel comfortable giving this to students!

File: projects/module-3/showcase.md [hunk 3]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
-* **1:**
-  - Many tests are missing/failing
-  - Tests do not successfully illustrate user flows
-  - No async functionality is stubbed
-  - There are obvious, large gaps in testing app functionality
+💫ON TRACK💫 can look like:
+* A consistent, modular file structure is used
+* A clear understanding of stateful components vs stateless components is demonstrated
+* Only the data that a child component _needs_ is passed down as props
+* Logic is kept out of return statements; return statements are as readable as possible, only communicating what will be displayed
```
Comment (hfaerber) [inline projects/module-3/showcase.md]: Can we add 'when possible' here?  Students saw this verbiage this inning took it as gospel rather than "pull out messy logic when it makes sense"

Logic is kept out of return statements <when possible>; return statements are as readable as possible, only communicating what will be displayed

File: projects/module-3/showcase.md [hunk 4]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
-  - README does not document your learning goals, the evolution of the project, and is poorly formatted (hindering readability)
-  - README does not include links to team member's GitHub profiles
-  - Git commits are not atomic and document changesets that are too large
-  - PRs do not have thorough descriptions, and no code reviews are conducted, merging bugs into the main branch
-  - When the project is run locally, the terminal shows errors and more than 5 warnings -->
+  - UX is clear and set up so the user has access to previous routes
+
+✨WOW✨ can look like:
+- Code was refactored to remove unnecessary or old code artifacts
+- When the user enters a bad URL, they are given an easy way to navigate to a working page
```
Comment (hfaerber) [inline projects/module-3/showcase.md]: I think I'd like to see having a way to navigate home on bad routes as an On Track rather than a WOW.   
Can we remove the WOW section altogether here - bringing line 195 to the On Track examples and deleting line 195 altogether (since they dont refactor router in for this project).

File: projects/module-3/showcase.md [hunk 5]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
+
+### Testing:
+
+💫ON TRACK💫 can look like:
+* All user flows are tested
+* All network request functionality is intercepted and stubbed
+* Tests make specific assertions about the content DOM elements contain
+
+✨WOW✨ can look like:
```
Comment (hfaerber) [inline projects/module-3/showcase.md]: Since we want students to at least try to test for sad paths too, can we include that in the On Track section and just not have a WOW section here?   Thats how we approached things in Mod 2 projects.  Sometimes there isn't a specific WOW - they would just earn a Wow but doing an really impressive and thorough job on the On Track items.   We have verbiage to that effect in the M2 projects as well

File: projects/module-3/showcase.md [hunk 6]
```diff
@@ -141,155 +142,82 @@ Project evals will be asynchronous.  The feedback for this project will be limit
...
+You will not receive a score for professionalism, but you should be meeting these expectations:
+💫ON TRACK💫 can look like:
+* Project is deployed
+* README concisely communicates the team's individual and joint learning goals, the evolution of the project, and team member reflections while using good formatting to enhance readability
+* README links to all user GitHub profiles and any applicable repos & deployed sites
+* Setup instructions for any and all repos are thorough and verbose enough that even non-technical people (like recruiters) could follow them
+* Git commits are atomic, with concise and precise descriptions of the change made
+* PRs have full, consistent descriptions
+* Team members conduct consistent code reviews of PRs
+* Evolution of the project (decisions made, etc) is documented in the git history and PRs
```
Comment (hfaerber) [inline projects/module-3/showcase.md]: Can we rephrase this to make wording fit for a solo project rather than team project?

File: projects/module-3/showcase.md (hunk 1 above)
Comment (hfaerber) [inline projects/module-3/showcase.md]: I'm trying to think how we can make it super clear here that they should do this *After* finishing their work - that we don't actually want to see class-components in their final product

Maybe:
* After your project is completely finished, you could deepen your understanding of React and strive for a WOW by creating a new branch and refactoring your stateful components into class-based components.  You will *not* merge this branch in.  Instead, DM a link for that branch/PR to your project manager.  Highlight your ability to work with both styles by describing the refactor and linking to the branch in your README.

File: projects/module-3/showcase.md (hunk 2 above)
Comment (hfaerber) [reply projects/module-3/showcase.md]: Can we also add the verbiage  below that we used in M2 to walk through the rubric scoring?

Project Requirements Rubric
For the rubric sections below, you will be scored as Wow, Yes or Not Yet depending on whether you have demonstrated competency in that area. Each section lists examples of what types of things we may be looking for as demonstrations of competency. Just as there are many ways to approach code, there are many many ways to demonstate competency. These are just some examples.

### Does the project demonstrate student understanding of the learning goals & concepts?
Projects will answer that question, being marked as yes, not yet, and wow. Similarly, each section of the rubric (see below) will have yes/not yet/wow markings, helping you understand your progress and growth in specific areas.

The overall project outcome (yes, not yet, wow) is determined by “averaging” each section’s outcome. You can think of a “yes” being worth a 1, a “not yet” being worth a 0, and a “wow” being worth a 2.

For this project, an average of 0.5 is considered a yes - a passing project that demonstrates good student understanding! An average of 1+ is considered a wow. Anything below a 0.5 is considered a not yet - a project that indicates that the concepts have not been fully understood (see note in the section below).

While M3 rubrics do not have a separate section for WOWs like in M1, there are a few WOW examples noted throughout. In addition to these WOW bullets, you can strive for a WOW by demonstrating not just competency, but excellence and thoroughness across the rubric sections.

**Review by hfaerber — CHANGES_REQUESTED** [review]
Thanks for your awesome work on this Scott!  

- Can we remove/rephrase the sections that reference teammates since this is a solo project?
- Can we add the breakdown of rubric scoring that we use in M2 projects?  I copied it into a comment.
- Can we remove the separate WOW sections to format the M3 rubrics more similarly to the M2 rubrics?  Where we outline examples for them to strive for and give them a WOW based on how excellently they do those things?  In cases were there is a specific extra thing for a WOW, we could note it how we did in the JS rubric section of [this m2 project](https://frontend.turing.edu/projects/module-2/whats-cookin-part-two-agile.html)
